Skip test requiring rpmdb

carlsmedstad opened the pull request:
Hey again! :wave:

A test was added without the proper predicate in the 2.1.0 release which causes it to fail on Arch Linux.

The added CI job should make sure this does not happen anymore in the future.

Conan-Kudo commented on .github/workflows/main.yml:
```diff
@@ -7,7 +7,45 @@ on:
+              python3-pip
+              rpm-build
+              git
+      - run: rm -rf $(rpm --eval %_dbpath)
```
Please quote `%_dbpath` here so that the shell doesn't evaluate it.

carlsmedstad replied:
Fixed!

marxin reviewed (changes requested):
As mentioned, I prefer using matrix with `container` and `build-type` properties, where the later can be one of `normal`, `optional-deps`.

carlsmedstad commented:
> As mentioned, I prefer using matrix with `container` and `build-type` properties, where the later can be one of `normal`, `optional-deps`.

Something like in https://github.com/rpm-software-management/rpmlint/pull/690/commits/3cfe85ba58f2cd077c2a7111149b2f2df9326845?

marxin commented on .github/workflows/main.yml:
```diff
@@ -17,6 +17,9 @@ jobs:
           - 'registry.opensuse.org/opensuse/tumbleweed-dnf:latest'
+        build-type:
+          - 'normal'
+          - 'no-optional-deps'
```
Nice! I would not run all 2x4 combinations, but I would rather use `include` and run only 2 `no-optional-deps` (one fedora and one opensuse).

marxin commented on .github/workflows/main.yml:
```diff
@@ -52,26 +52,26 @@ jobs:
+              appstream-glib
+              myspell-en_US myspell-cs_CZ
+              python3-pyenchant
+        if: ${{ contains(matrix.docker, 'opensuse') && contains(matrix.build-type, 'normal') }}
```
Use rather `matrix.build-type == 'normal'`, please.

marxin commented on .github/workflows/main.yml:
```diff
@@ -88,6 +88,17 @@ jobs:
+              hunspell-en
+              hunspell-cs
+              python3-enchant
+        if: ${{ contains(matrix.docker, 'fedora') && contains(matrix.build-type, 'normal') }}
```
Likewise here.

marxin reviewed (changes requested):
Very nice improvement, please fix the remaining comments I have.

carlsmedstad commented:
I believe I've addressed all the comments. Also, I've squashed any redundant commits.

OpenSUSE's docker registry seems to be down thought, so those tests are failing.

marxin commented:
> I believe I've addressed all the comments. Also, I've squashed any redundant commits.

Great, thank you for the contribution.
